numba/array_analysis.py

```diff
@@ -1422,7 +1432,59 @@ def _analyze_op_call_numpy_reshape(self, scope, equiv_set, args, kws):
             typ = self.typemap[args[1].name]
             if isinstance(typ, types.BaseTuple):
                 return args[1], []
-        return tuple(args[1:]), []
+
+        # Reshape is allowed to take one argument that has the value -1.
```

Comment:
I'm wondering whether this belongs in analysis given it's mutating IR?

Reply:
That's up to you but there's a lot more rewriting going on than what I added for reshape.  Variables and assignment for array dimension size calculation and storage are being inserted.

Reply:
ok, let's just leave it for now.